### ISSUE TITLE
no-disabled-tests doesn't recognice spreading

### ISSUE THREAD

**Issue by adrianjost** [jest-community/eslint-plugin-jest#518]
We have defined a couple of tests that we run for a lot of components we build. Those tests get auto generated by a method and the return value gets spreaded into the test method. Unfortunatly the rule `no-disabled-tests` can not resolve the spreading and therefore throws an error ( `Test is missing function argument eslint(jest/no-disabled-tests)` ).

minimal example:
```js
const test = [
	"title",
	() => { expect(true).toBe(true); },
];
describe("@components/someComponent", () => {
	it(...test); // warning  Disabled test suite          jest/no-disabled-tests
})
```

a more practial example [would be this](https://github.com/schul-cloud/nuxt-client/blob/develop/src/components/atoms/PulsatingDot.unit.js):
```js
import PulsatingDot from "./PulsatingDot";
describe("@components/atoms/PulsatingDot", () => {
	// isValidComponent returns 
	// ["PulsatingDot exports a valid component", () => { ... }]
	it(...isValidComponent(PulsatingDot));
});
``` 

Is there an easy way to fix this on our side or is the rule just not compatible with this approach?

---

Research I did:
- The error seems to be coming from [here](https://github.com/jest-community/eslint-plugin-jest/blob/master/src/rules/no-disabled-tests.ts#L36) and is thrown because the spreading isn't detected as multiple arguments. Maybe there is another method to check the number of parameters?

**Comment by G-Rath** [comment]
@adrianjost I should be able to fix this, tho I'd recommend you use index accessing rather than spreading the argument

(Just an FYI this isn't destructing, it's spreading)

**Comment by adrianjost** [comment]
@G-Rath That would be awesome. 
And sorry, I mix these words sometimes. Shame on me 😌 I will adjust the issue for easier reference in the future.

Could you give me an example how to use index accessing in this case? I think it would be pretty awkward to write something like `it(test[0], test[1]);` in every testcase.

**Comment by G-Rath** [comment]
@adrianjost I thought that you said the tests got auto generated?

Hence have the generator generate the index access, or better yet the whole test?

**Comment by adrianjost** [comment]
@G-Rath Yes kindof. We have a global function called `isValidComponent` which will receive a Component as a parameter and uses this component to generate and return an Array that includes the testname with the populated component name as the first entry and a test function populated with the component to test as the second parameter.

**Comment by G-Rath** [comment]
I'm going to close this at `wontfix` as I don't think it's a pattern we want to encourage.

You could easily work around this by changing `isValidComponent` to contain the `it`:

```
const isValidComponent = (...) => {
  it("title", (​)​ ​=>​ ​{​ ​expect​(​true​)​.​toBe​(​true​)​;​ ​}​):
}
```

IMO this is better anyway as it encapsulates the responsibility of the function.
